packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -140,34 +156,53 @@ contract SuperchainConfig is Initializable, ISemver {
         emit ConfigUpdate(UpdateType.GUARDIAN, abi.encode(_guardian));
     }
 
-    /// @notice Sets the dependency manager address. This is only callable during initialization, so an upgrade
-    ///         will be required to change the dependency manager.
-    /// @param _dependencyManager The new dependency manager address.
-    function _setDependencyManager(address _dependencyManager) internal {
-        Storage.setAddress(DEPENDENCY_MANAGER_SLOT, _dependencyManager);
-        emit ConfigUpdate(UpdateType.DEPENDENCY_MANAGER, abi.encode(_dependencyManager));
+    /// @notice Sets the cluster manager address. This is only callable during initialization, so an upgrade
+    ///         will be required to change the cluster manager.
+    /// @param _clusterManager The new cluster manager address.
+    function _setClusterManager(address _clusterManager) internal {
+        Storage.setAddress(CLUSTER_MANAGER_SLOT, _clusterManager);
+        emit ConfigUpdate(UpdateType.CLUSTER_MANAGER, abi.encode(_clusterManager));
     }
 
     /// @notice Adds a new dependency to the dependency set. It also authorizes it's OptimismPortal on the
-    ///         SharedLockbox. Can only be called by the dependency manager.
-    /// @param _chainId         The chain ID.
+    ///         SharedLockbox and migrate it's ETH liquidity to it. Can only be called by an authorized
+    ///         OptimismPortal via a withdrawal transaction initiated by the DependencyManager.
+    /// @param _chainId         The chain ID to add.
     /// @param _systemConfig    The SystemConfig contract address of the chain to add.
     function addDependency(uint256 _chainId, address _systemConfig) external {
-        if (msg.sender != dependencyManager()) revert Unauthorized();
+        if (msg.sender != clusterManager()) {
+            if (!authorizedPortals[msg.sender]) revert Unauthorized();
+            if (IOptimismPortal2(payable(msg.sender)).l2Sender() != Predeploys.DEPENDENCY_MANAGER) {
+                revert Unauthorized();
+            }
+        }
 
         if (_dependencySet.length() == type(uint8).max) revert DependencySetTooLarge();
-        if (_chainId == block.chainid) revert InvalidChainID();
+        if (_chainId == block.chainid) revert InvalidChainID(); // TODO: is this check really necessary?
```

Comment:
This check is good for preventing footguns altho i don't think its that useful in practice. If we mess up the config for the dep set then we are going to have a bad time regardless of this check. I would be down to remove this to reduce test maintenance